src/vdaf/prio3.rs

```diff
@@ -249,13 +253,10 @@ pub struct Prio3InputShare<F> {
     pub proof_share: Share<F>,
 
     /// The sum of the joint randomness seed shares sent to the other aggregators.
-    //
-    // TODO(cjpatton) If `input.joint_rand_len() == 0`, then we don't need to bother with the joint
-    // randomness seed at all. See https://github.com/cjpatton/vdaf/issues/15.
-    pub joint_rand_seed_hint: Key,
+    pub joint_rand_seed_hint: Option<Key>,
 
     /// The blinding factor, used to derive the aggregator's joint randomness seed share.
-    pub blind: Key,
+    pub blind: Option<Key>,
```

Comment:
This allows the possibility of a `Prio3InputShare` with `joint_rand_seed_hint = None` and `blind = Some(...)`, or vice versa, which are both invalid and shouldn't be allowed. `Prio3InputShare::decode` should check for this and reject such values, or we could define a field `joint_rand_seed_hint_and_blind: Option<(Key, Key)>` so that they must either both be set or neither is set. Even better would be to define another struct that contains the `joint_rand_seed_hint` and `blind` but I'm not sure what to name it.

Reply:
Done (sorry about the rebase + force-push, but you can see the changes in the last commit ("SQUASH Tim round 1")